### ISSUE TITLE
A lesson on shell / bash setup

### ISSUE THREAD

**Issue by lwasser** [pyOpenSci/python-package-guide#134]
In our guide - we spent a bit of time discussing users and various shells. 
[here is the conversation](https://github.com/pyOpenSci/python-package-guide/pull/106#discussion_r1428339852)

in my experience teaching this topic, many don't know / understand bash / shell / windows terminal etc and as such this could become a stumbling block. It could be worth us having a page somewhere that helps users understand what shell they are using and associated setup items that they may need to consider. 

tagging @ucodery and @sneakers-the-rat as they were heavily involved in this conversation as well as @Zeitsperre

**Comment by sneakers-the-rat** [comment]
This is where we start getting into wiki territory :) so ideally it would be easy for us to drop in pages for lots of different topics as they come up. In wikis this is done with wikilinks, but in Sphinx style docs this could be maybe more like using the `index` directive - the places where eg. Shells are mentioned go into the index, and then we can manage links between places in guides where topics are mentioned and longer dedicated pages for those topics.

One of the reasons wikis are great is they really lower the barrier to making additions like this, so we might think of having an adjoining wiki for extra topics like these so we can just make pages as-needed :). But in any case I feel like for now another document thats more like "environment management and DX" stuff would be great - IDEs and the practicalities of interacting with code are a huge grey space in documentation, as I am learning again when teaching a student from scratch.

I dont feel like we should try and document the whole universe (but we can try!) So even something as little as gathering links out to other resources by topic would be great pending a Fuller docs push

**Comment by ucodery** [comment]
I agree that a fair amount of our readers likely won't know what shell they're using, or what a shell is (or a terminal, pty). It would be nice to have a page to introduce them, or to at least let them answer the question. But this also seems pretty far afield from packaging.

This reminds me that I once saw @b0rk@jvns.ca state that the best way to determine the currently-running shell was to execute a nonsense command (like `this-will-never-work-its-not-a-command`) and then read the error message, as most shells report their name in the error message.

**Comment by lwasser** [comment]
This is an old-ish issue, but still potentially valid, especially as we run sprints. I am going to come back to it and add labels. I know this isn't about packaging, but I do think shell as a topic is important and confusing for people. And the commands can be slightly different.

**Comment by ucodery** [comment]
Food for thought: would this be better as a self-contained tutorial rather than appearing in the middle of a Packaging Guide, where it is slightly non sequitur?
